src/CoreBundle/private/library/classes/TCMSFields/TCMSFieldExtendedLookup.class.php

```diff
@@ -152,6 +150,30 @@ protected function _GetOpenWindowJS(&$oPopupTableConf)
         return $js;
     }
 
+    private function getTargetListRestriction(): array
+    {
+        $restrictionExpression = $this->oDefinition->GetFieldtypeConfigKey('restriction');
+        if (null === $restrictionExpression) {
+            return [];
+        }
+        if (false === \strpos($restrictionExpression, '=')) {
+            return [];
+        }
+        $equalPosition = \mb_strpos($restrictionExpression, '=');
+        $restrictionField = trim(\mb_substr($restrictionExpression, 0, $equalPosition));
+        $restrictionValue = trim(\mb_substr($restrictionExpression, $equalPosition+1));
+        $restrictionValue = mb_substr($restrictionValue, 2, -2); // ignore [{}]
+        if (\property_exists($this, 'oTableRow') && \is_object($this->oTableRow) && \property_exists($this->oTableRow, 'sqlData') && is_array($this->oTableRow->sqlData)) {
+            if (isset($this->oTableRow->sqlData[$restrictionValue])) {
+                $restrictionValue = $this->oTableRow->sqlData[$restrictionValue];
+            }
+        }
+        return [
+            'sRestrictionField' => $restrictionField,
+            'sRestriction' => $restrictionValue,
```

Comment:
(old problem) $restrictionValue (or $restrictionField) could still be empty here.

Reply:
ok - check added